crates/oro-node-semver/src/version_req.rs

```diff
@@ -153,38 +153,49 @@ where
+where
+    E: ParseError<&'a str>,
+{
+    tuple((number, maybe_dot_number, maybe_dot_number))(input)
+}
+
+fn maybe_dot_number<'a, E>(input: &'a str) -> IResult<&'a str, Option<u64>, E>
+where
+    E: ParseError<&'a str>,
+{
+    opt(preceded(tag("."), number))(input)
+}
+
 fn any_operation_followed_by_version<'a, E>(input: &'a str) -> IResult<&'a str, Range, E>
 where
     E: ParseError<&'a str>,
 {
     context(
         "operation followed by version",
-        map(
-            tuple((operation, number, maybe_dot_number, maybe_dot_number)),
-            |parsed| match parsed {
-                (Operation::GreaterThanEquals, major, minor, None) => Range::Open(Predicate {
-                    operation: Operation::GreaterThanEquals,
-                    version: (major, minor.unwrap_or(0), 0).into(),
-                }),
-                (Operation::GreaterThan, major, Some(minor), None) => Range::Open(Predicate {
-                    operation: Operation::GreaterThanEquals,
-                    version: (major, minor + 1, 0).into(),
-                }),
-                (Operation::GreaterThan, major, None, None) => Range::Open(Predicate {
-                    operation: Operation::GreaterThanEquals,
-                    version: (major + 1, 0, 0).into(),
-                }),
-                (Operation::LessThan, major, minor, None) => Range::Open(Predicate {
-                    operation: Operation::LessThan,
-                    version: (major, minor.unwrap_or(0), 0).into(),
-                }),
-                (operation, major, Some(minor), Some(patch)) => Range::Open(Predicate {
-                    operation,
-                    version: (major, minor, patch).into(),
-                }),
-                _ => panic!("Unexpected"),
-            },
-        ),
+        map(tuple((operation, partial_version)), |parsed| match parsed {
+            (Operation::GreaterThanEquals, (major, minor, None)) => Range::Open(Predicate {
+                operation: Operation::GreaterThanEquals,
+                version: (major, minor.unwrap_or(0), 0).into(),
+            }),
+            (Operation::GreaterThan, (major, Some(minor), None)) => Range::Open(Predicate {
+                operation: Operation::GreaterThanEquals,
+                version: (major, minor + 1, 0).into(),
+            }),
+            (Operation::GreaterThan, (major, None, None)) => Range::Open(Predicate {
+                operation: Operation::GreaterThanEquals,
+                version: (major + 1, 0, 0).into(),
+            }),
+            (Operation::LessThan, (major, minor, None)) => Range::Open(Predicate {
+                operation: Operation::LessThan,
+                version: (major, minor.unwrap_or(0), 0, 0).into(),
+            }),
+            (operation, (major, Some(minor), Some(patch))) => Range::Open(Predicate {
+                operation,
+                version: (major, minor, patch).into(),
+            }),
+            _ => panic!("Unexpected"),
```

Comment:
You can use the `unreachable!()` macro here btw

Reply:
Oh yeah! `\o/`